<launch>
    <arg name="debug" default="false"/>
    <arg name="fullscreen" default="true"/>
    <arg name="start_delay" default="0.0" />
    <arg name="rviz" default="true"/>
    <arg name="args" default=""/>
    <arg name="sleep" default="bash -c 'sleep $(arg start_delay); $0 $@' "/>

    <arg unless="$(arg debug)" name="launch-prefix" value="$(arg sleep)"/>
    <arg     if="$(arg debug)" name="launch-prefix" value="$(arg sleep) gdb -ex run --args"/>

    <arg unless="$(arg fullscreen)" name="rviz-fullscreen" value=""/>
    <arg     if="$(arg fullscreen)" name="rviz-fullscreen" value="--fullscreen"/>

    <!-- robot description -->
    <param name="robot_description" command="cat '$(find fetch_description)/robots/fetch.urdf'"/>

    <!-- 0. RViz -->
    <node if="$(arg rviz)" type="rviz" name="rviz" pkg="rviz" args="-d $(find smpl_test)/config/config.rviz $(arg rviz-fullscreen)"/>

    <!-- 1. call planner -->
    <node pkg="smpl_test" type="callPlannerBenchmarking" name="call_planner_benchmarking" launch-prefix="$(arg launch-prefix)" args="$(arg args)">
        <!-- robot model (for planning) -->
        <rosparam param="robot_model">
            kinematics_frame:
                base_link
            chain_tip_link:
                gripper_link
        </rosparam>

        <!-- collision checking -->
        <rosparam command="load" file="$(find sbpl_collision_checking_test)/config/collision_model_fetch.yaml"/>

        <!-- planner parameters -->
        <rosparam command="load" file="$(find smpl_test)/config/fetch_arm.yaml"/>
        <!-- TODO: check linear motion primitives' units -->
        <param name="planning/mprim_filename" value="$(find smpl_test)/config/fetch.mprim"/>

        <param name="planning_problems_directory" value="$(find robowflex_library)/problems/fetch/bookshelf_small_fetch"/>
        <param name="planning_problem_index_start" value="1"/>
        <param name="planning_problem_index_end" value="1"/>
        <param name="planning_problem_index_width" value="4"/>

        <param name="planning_algorithm" value="arastar"/>  <!-- options: arastar, epase -->
        <param name="goal_type" value="joints"/>  <!-- options: joints, pose -->
        <param name="num_threads" value="4"/>
        <param name="planning_frame" value="odom_combined"/>

        <param name="verbose" value="true"/>
        <param name="visualize" value="true"/>  <!-- TODO: depend on rviz arg -->
    </node>
</launch>
